lib/poller.js

```diff
@@ -74,49 +94,68 @@ class Poller {
    * @returns {Promise} Promise object that always resolves.
    */
   async _poll () {
-    this._logger.info(`running poll on the secret ${this._secretDescriptor.name}`)
+    this._logger.info(`running poll on the secret ${this._name}`)
+
     try {
       await this._upsertKubernetesSecret()
+
+      this._metrics.observeSync({
+        name: this._name,
+        namespace: this._namespace,
+        backend: this._secretDescriptor.backendType,
+        status: 'success'
+      })
     } catch (err) {
-      this._logger.error(err, `failure while polling the secret ${this._secretDescriptor.name}`)
+      this._logger.error(err, `failure while polling the secret ${this._name}`)
       this._metrics.observeSync({
-        name: this._secretDescriptor.name,
+        name: this._name,
         namespace: this._namespace,
         backend: this._secretDescriptor.backendType,
         status: 'error'
       })
+      await this._updateStatus(`ERROR, ${err.message}`)
     }
-
-    this._metrics.observeSync({
-      name: this._secretDescriptor.name,
-      namespace: this._namespace,
-      backend: this._secretDescriptor.backendType,
-      status: 'success'
-    })
   }
 
   /**
    * Create or update Kubernets secret in the cluster.
    * @returns {Promise} Promise object representing operation result.
    */
   async _upsertKubernetesSecret () {
-    const secretDescriptor = this._secretDescriptor
     const kubeNamespace = this._kubeClient.api.v1.namespaces(this._namespace)
 
     // check if namespace is allowed to fetch this secret
     const ns = await kubeNamespace.get()
-    const verdict = this._isPermitted(ns.body, secretDescriptor)
+    const verdict = this._isPermitted(ns.body, this._secretDescriptor)
+
     if (!verdict.allowed) {
-      throw (new Error(`not allowed to fetch secret: ${secretDescriptor.name}: ${verdict.reason}`))
+      throw (new Error(`not allowed to fetch secret: ${this._name}: ${verdict.reason}`))
     }
+
     const secretManifest = await this._createSecretManifest()
-    this._logger.info(`upserting secret ${this._secretDescriptor.name} in ${this._namespace}`)
+    this._logger.info(`upserting secret ${this._name} in ${this._namespace}`)
+
     try {
-      return await kubeNamespace.secrets.post({ body: secretManifest })
+      await kubeNamespace.secrets.post({ body: secretManifest })
     } catch (err) {
       if (err.statusCode !== 409) throw err
-      return kubeNamespace.secrets(this._secretDescriptor.name).put({ body: secretManifest })
+      await kubeNamespace.secrets(this._name).put({ body: secretManifest })
     }
+
+    await this._updateStatus('SUCCESS')
```

Comment:
nit: move this to `._updateStatus` to https://github.com/godaddy/kubernetes-external-secrets/pull/133/files#diff-ef923d397aa8d0a9b6bc6d401b5290d7R101.  Then the 'SUCCESS' and 'ERROR' status setting are in the same function. When I was reading this the first time I aw the 'ERROR' status and wondered why there wasn't a 'SUCCESS' status near by.